### PR TITLE
fixes #5634 - save sso_method on session expiry

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

**Comment by stbenjam** [comment]
Need to add tests too, there's not much functional coverage of the authorization stuff.

File: app/controllers/concerns/foreman/controller/session.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def update_activity_time
 
   def expire_session
     logger.info "Session for #{User.current} is expired."
-    reset_session
+    backup_session_content { reset_session }
```
Comment (domcleal) [inline app/controllers/concerns/foreman/controller/session.rb:30]: This is done already in `session_expiry`, where's it being called from that doesn't wrap in backup_session_content?  (The exception handler?)

**Comment by domcleal** [comment]
Try adding to test/functional/application_controller_subclass_test.rb.

File: app/controllers/concerns/foreman/controller/session.rb (hunk 1 above)
Comment (stbenjam) [inline app/controllers/concerns/foreman/controller/session.rb:30]: Ah, I see.  Wouldn't it be better to remove it from session_expiry and leave the wrapped reset_session call here?

Comment (domcleal) [inline app/controllers/concerns/foreman/controller/session.rb:30]: Yes, quite possibly, not sure why I did that...

But I'd like to understand how expire_session's getting called, particularly if it was an exception being thrown.

Comment (stbenjam) [inline app/controllers/concerns/foreman/controller/session.rb:30]: It's not the exception case, what happens is, you've wrapped all of expire_session in the backup.  

Session details aren't restored until _after_ the redirect.

Comment (domcleal) [inline app/controllers/concerns/foreman/controller/session.rb:30]: Ahh, gotcha.  :+1: then

**Comment by dLobatog** [comment]
@stbenjam Aside from the tests this needs a rebase against develop

**Comment by dLobatog** [comment]
@stbenjam ping

**Comment by stbenjam** [comment]
@eLobato pong :cactus:

**Comment by dLobatog** [comment]
Merged as fae8b987a8fb329d3cc6a9243e967861d7f5d198 , thanks @stbenjam, 6 months to get this fix in, sorry :bow: !

**Pull request closed without merging**